utils.go

```diff
@@ -0,0 +1,49 @@
+/*----------------------------------------------------
+ *
+ *  2022 NVIDIA CORPORATION & AFFILIATES
+ *
+ *  Licensed under the Apache License, Version 2.0 (the License);
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an AS IS BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *----------------------------------------------------
+ */
+
+package sriovnet
+
+import (
+	"fmt"
+	"strings"
+
+	utilfs "github.com/Mellanox/sriovnet/pkg/utils/filesystem"
+)
+
+func getFileNamesFromPath(dir string) ([]string, error) {
+	_, err := utilfs.Fs.Stat(dir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get network devices from path %s: %v", dir, err)
+	}
+
+	netDevicesFiles, err := utilfs.Fs.ReadDir(dir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get network devices name in %s: %v", dir, err)
+	}
+
+	if len(netDevicesFiles) == 0 {
```

Comment:
getting files from an empty dir is valid and should return an empty list. if the caller expects non empty output it should enforce it.